### ISSUE TITLE
`jnp.svd` intermittently ignores `compute_uv=False`

### ISSUE THREAD

**Issue by Michael-T-McCann** [google/jax#9483]
We've been having intermittent failures in our CI over at https://github.com/lanl/scico, and I believe I have tracked the cause down to some odd behavior in `jnp.svd`. The assert in the following code fails about one in one hundred calls.

```python
import jax
import jax.numpy as jnp


def test():
    key = jax.random.PRNGKey(3)
    v = jax.random.normal(shape=(32, 1), key=key)

    svdU, svdS, svdV = jnp.linalg.svd(v, full_matrices=False)
    x = svdU @ jnp.diag(svdS) @ svdV
    sigma = jnp.linalg.svd(x, compute_uv=False)
    assert len(sigma) == 1

test()
```
The bash loop I'm using to test is
```bash
while python main.py
do
done
```

I have not been able to get a failure if I remove the first `svd`, but I may have just been lucky/unlucky. I've seen the failures both on my MacOS laptop and on the github CI Linux machines. This is on jax 0.2.28 with jaxlib 0.1.76, CPU version.

Any ideas on why this might happen, work arounds, or even ideas to help provoke the error more frequently would be appreciated!

**Comment by jakevdp** [comment]
Thanks for the report – that is strange! I was able to reproduce it in a Colab CPU runtime after about 500 iterations with the following 3 cells:
```python
pip install jax==0.2.28 jaxlib==0.1.76
```
```python
%%writefile main.py
import jax
import jax.numpy as jnp

def test():
    key = jax.random.PRNGKey(3)
    v = jax.random.normal(shape=(32, 1), key=key)

    svdU, svdS, svdV = jnp.linalg.svd(v, full_matrices=False)
    x = svdU @ jnp.diag(svdS) @ svdV
    sigma = jnp.linalg.svd(x, compute_uv=False)
    assert len(sigma) == 1

test()
```
```python
import subprocess
for i in range(1000):
  if i % 10 == 0: print(i)
  out = subprocess.run(['python3', 'main.py'], capture_output=True)
  if out.returncode:
    print(out.stderr.decode())
    break
```

**Comment by jakevdp** [comment]
I was wondering if this might be an issue with our static argument cacheing, so I tried to replicate this *without* a call to `svd` with these Colab notebook cells:
```python
%%writefile main2.py
from functools import partial
from jax import jit
import jax.numpy as jnp

@partial(jit, static_argnames=['ravel', 'make_tuple'])
def func(x, ravel=True, make_tuple=True):
  if ravel:
    x = x.ravel()
  return (x, x) if make_tuple else x

def test():
  x = jnp.ones((3, 3))
  y, _ = func(x, ravel=False)
  out = func(y, make_tuple=False)
  assert len(out) == 9

test()  
```
```python
import subprocess
for i in range(1000):
  if i % 10 == 0: print(i)
  out = subprocess.run(['python3', 'main2.py'], capture_output=True)
  if out.returncode:
    print(out.stderr.decode())
    break
```
I ran this a couple times and didn't seem to reproduce the error. This makes me think the bug is related to the SVD implementation, rather than something related to JIT dispatch.

**Comment by zhangqiaorjc** [comment]
We repro'ed internally on CPU backend, will follow up with findings.

cc @hawkinsp

**Comment by jakevdp** [comment]
Thanks for the report! The repro was very helpful. The issue should be fixed by https://github.com/tensorflow/tensorflow/commit/c5c390ae36fab842a02bb171f3268c3faef84151 once that makes it into a jaxlib build.

**Comment by hawkinsp** [comment]
We released `jax` and `jaxlib` 0.3.0, which includes the fix for this issue.
